cli/src/commands/config/edit.rs

```diff
@@ -41,5 +42,17 @@ pub fn cmd_config_edit(
         file.save()?;
     }
     editor.edit_file(file.path())?;
+
+    // Trying to load back config. If error, saves the old config and warn the user.
+    if let Err(e) = ConfigLayer::load_from_file(file.layer().source, file.path().to_path_buf()) {
+        writeln!(
+            ui.error_with_heading("Error: "),
+            "An error has been found inside your config. Restoring previous one."
+        )?;
```

Comment:
nit: use `user_error_with_message()` or something to error out. (or use `ui.warning_default()` to print warning and continue.)